### PR TITLE
CSPL-2125: operator-sdk and golang upgrade

### PR DESCRIPTION
**Pull request by gaurav-splunk**
Updated operator-sdk to `v1.25.0` and golang to `v1.19.2`.

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -16,5 +16,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: splunk/splunk-operator
-  newTag: 2.1.0
+  newName: docker.io/gaurav23gupta/splunk-operator
```
Comment (kumarajeet) [inline config/manager/kustomization.yaml:19]: is that name intended?

File: test/appframework_aws/s1/appframework_aws_test.go [hunk 2]
```diff
@@ -1724,9 +1723,6 @@ var _ = Describe("s1appfw test", func() {
 			Expect(err).To(Succeed(), fmt.Sprintf("Unable to upload %s apps to S3 test directory for Standalone", appVersion))
 			uploadedApps = append(uploadedApps, uploadedFiles...)
 
-			// Wait for Standalone to be in READY status
-			testenv.StandaloneReady(ctx, deployment, deployment.GetName(), standalone, testcaseEnvInst)
```
Comment (kumarajeet) [inline test/appframework_aws/s1/appframework_aws_test.go:1728]: just checking why this line was removed?

File: Makefile [hunk 3]
```diff
@@ -138,6 +138,23 @@ docker-build: test ## Build docker image with the manager.
 docker-push: ## Push docker image with the manager.
 	docker push ${IMG}
 
+# PLATFORMS defines the target platforms for  the manager image be build to provide support to multiple
```
Comment (kumarajeet) [inline Makefile:141]: was this mandatory for the upgrade?

File: config/manager/kustomization.yaml (hunk 1 above)
Comment (gaurav-splunk) [reply config/manager/kustomization.yaml:19]: good catch, will revert it

File: .github/workflows/build-test-push-workflow.yml [hunk 4]
```diff
@@ -185,17 +166,14 @@ jobs:
...
-          sudo ./awscli-bundle/install -i /usr/local/aws -b /usr/local/bin/aws
+          curl "${{ steps.dotenv.outputs.AWSCLI_URL}}" -o "awscliv2.zip"
+          unzip awscliv2.zip
+          sudo ./aws/install --update
           aws --version
       - name: Setup Go
         uses: actions/setup-go@v2
         with:
           go-version: ${{ steps.dotenv.outputs.GO_VERSION }}
-      - name: Install Go Lint
```
Comment (kumarajeet) [inline .github/workflows/build-test-push-workflow.yml:196]: assuming not needed?

File: .github/workflows/int-test-workflow.yml [hunk 5]
```diff
@@ -4,6 +4,7 @@ on:
     branches:
       - develop
       - main
+      - prodsec-fix
```
Comment (kumarajeet) [inline .github/workflows/int-test-workflow.yml:7]: will we continue to include prodsec-fix branch in the tests?

File: test/appframework_aws/s1/appframework_aws_test.go (hunk 2 above)
Comment (gaurav-splunk) [reply test/appframework_aws/s1/appframework_aws_test.go:1728]: I think @tpereirasplunk made this change to fix failing S1 test case

File: Makefile (hunk 3 above)
Comment (gaurav-splunk) [reply Makefile:141]: Yeah it was part of the operator-sdk upgrade

File: .github/workflows/build-test-push-workflow.yml (hunk 4 above)
Comment (gaurav-splunk) [reply .github/workflows/build-test-push-workflow.yml:196]: yup, support for golint is deprecated with the latest versions.

File: .github/workflows/int-test-workflow.yml (hunk 5 above)
Comment (gaurav-splunk) [reply .github/workflows/int-test-workflow.yml:7]: once everything passses, will remove this line.

File: .env [hunk 6]
```diff
@@ -1,8 +1,8 @@
...
-GO_VERSION=1.17.3
-AWSCLI_URL=https://s3.amazonaws.com/aws-cli/awscli-bundle-1.20.8.zip
-KUBECTL_VERSION=v1.23.6
+GO_VERSION=1.19.2
+AWSCLI_URL=https://awscli.amazonaws.com/awscli-exe-linux-x86_64-2.8.6.zip
+KUBECTL_VERSION=v1.25.3
 AZ_CLI_VERSION=2.30.0
-EKSCTL_VERSION=v0.114.0
+EKSCTL_VERSION=v0.115.0
 EKS_CLUSTER_K8_VERSION=1.22
```
Comment (sgontla) [inline .env:8]: For all the version upgrades, i think we also need to address it in our docs?

Comment (sgontla) [reply .env:8]: NM, seems like you alrady addressed it :-)

File: bundle/manifests/splunk-operator.clusterserviceversion.yaml [hunk 7]
```diff
@@ -799,14 +802,14 @@ spec:
                     fieldRef:
                       fieldPath: metadata.annotations['olm.targetNamespaces']
                 - name: RELATED_IMAGE_SPLUNK_ENTERPRISE
-                  value: splunk/splunk:9.0.1-a1
+                  value: docker.io/splunk/splunk:edge
```
Comment (sgontla) [inline bundle/manifests/splunk-operator.clusterserviceversion.yaml:805]: Do we need this change?

File: bundle/manifests/splunk-operator.clusterserviceversion.yaml [hunk 8]
```diff
@@ -911,7 +914,7 @@ spec:
     name: Splunk Inc.
     url: www.splunk.com
   relatedImages:
-  - image: splunk/splunk:9.0.1-a1
+  - image: docker.io/splunk/splunk:edge
```
Comment (sgontla) [inline bundle/manifests/splunk-operator.clusterserviceversion.yaml:917]: Same here.

File: bundle/manifests/splunk-operator.clusterserviceversion.yaml [hunk 9]
```diff
@@ -799,14 +802,14 @@ spec:
...
-                  value: splunk/splunk:9.0.1-a1
+                  value: docker.io/splunk/splunk:edge
                 - name: OPERATOR_NAME
                   value: splunk-operator
                 - name: POD_NAME
                   valueFrom:
                     fieldRef:
                       fieldPath: metadata.name
-                image: splunk/splunk-operator:2.1.0
+                image: controller:latest
```
Comment (gaurav-splunk) [inline bundle/manifests/splunk-operator.clusterserviceversion.yaml:812]: need to check with @vivekr-splunk if this is correct.

File: bundle/manifests/splunk-operator.clusterserviceversion.yaml [hunk 10]
```diff
@@ -911,7 +914,7 @@ spec:
     name: Splunk Inc.
     url: www.splunk.com
   relatedImages:
-  - image: splunk/splunk:9.0.1-a1
+  - image: docker.io/splunk/splunk:edge
     name: splunk-enterprise
   replaces: splunk-operator.v0.1.0
-  version: 2.1.0
+  version: 2.0.0
```
Comment (gaurav-splunk) [inline bundle/manifests/splunk-operator.clusterserviceversion.yaml:920]: same here, CC: @vivekr-splunk

**Review by pdhanoya-splunk — APPROVED** [review]

File: bundle/metadata/annotations.yaml [hunk 11]
```diff
@@ -4,8 +4,8 @@ annotations:
   operators.operatorframework.io.bundle.manifests.v1: manifests/
   operators.operatorframework.io.bundle.metadata.v1: metadata/
   operators.operatorframework.io.bundle.package.v1: splunk-operator
-  operators.operatorframework.io.bundle.channel.default.v1: "stable"
-  operators.operatorframework.io.metrics.builder: operator-sdk-v1.18.1
+  operators.operatorframework.io.bundle.channels.v1: alpha
```
Comment (vivekr-splunk) [inline bundle/metadata/annotations.yaml:7]: operators.operatorframework.io.bundle.channels.v1: "stable"
  operators.operatorframework.io.bundle.channel.default.v1: "stable"

File: bundle/manifests/splunk-operator.clusterserviceversion.yaml (hunk 7 above)
Comment (vivekr-splunk) [reply bundle/manifests/splunk-operator.clusterserviceversion.yaml:805]: this is an autogenerated file. changes need to be set to 9.1 splunk release

File: bundle/manifests/splunk-operator.clusterserviceversion.yaml (hunk 9 above)
Comment (vivekr-splunk) [reply bundle/manifests/splunk-operator.clusterserviceversion.yaml:812]: you can leave this as it is , i need to generate bundle file. once you merge the code, i will create the new PR just for bundle generation.

File: bundle/manifests/splunk-operator.clusterserviceversion.yaml (hunk 8 above)
Comment (vivekr-splunk) [reply bundle/manifests/splunk-operator.clusterserviceversion.yaml:917]: this is an autogenerated file. changes need to be set to 9.1 splunk release

File: bundle/manifests/splunk-operator.clusterserviceversion.yaml (hunk 10 above)
Comment (vivekr-splunk) [reply bundle/manifests/splunk-operator.clusterserviceversion.yaml:920]: bundle version changes to 2.1.0

File: bundle/manifests/splunk-operator.clusterserviceversion.yaml (hunk 7 above)
Comment (vivekr-splunk) [reply bundle/manifests/splunk-operator.clusterserviceversion.yaml:805]: we need to run 
`
make bundle IMAGE_TAG_BASE=docker.io/splunk/splunk-operator VERSION=2.1.0 IMG=splunk/splunk-operator:2.1.0 SPLUNK_ENTERPRISE_IMAGE=splunk/splunk:9.1
`

File: bundle/manifests/enterprise.splunk.com_clustermanagers.yaml [hunk 12]
```diff
@@ -2,7 +2,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.7.0
+    controller-gen.kubebuilder.io/version: v0.9.0
```
Comment (vivekr-splunk) [inline bundle/manifests/enterprise.splunk.com_clustermanagers.yaml:5]: when I generate it adds `x-kubernetes-map-type: atomic` for maps in CRD. i didnt see those changes here. also my kubebuilder version is 0.9.2

**Review by vivekr-splunk — APPROVED** [review]
i have updated CRD. so everything looks ok

**Pull request merged**